p2p/src/net/default_backend/backend.rs

```diff
@@ -523,63 +571,64 @@ where
                     boxed_cb(move |this| this.handle_connect_res(address, connection_res))
                 }
-                .boxed()
+                .boxed();
+
+                self.command_queue.push(backend_task);
             }
-            Command::Disconnect { peer_id } => async move {
-                boxed_cb(move |this: &mut Self| {
-                    let res = this.disconnect_peer(peer_id);
-                    if let Err(e) = res {
-                        log::debug!("Failed to disconnect peer {peer_id}: {e}")
-                    }
-                    Ok(())
-                })
+            Command::Accept { peer_id } => {
+                let res = self.accept_peer(peer_id);
+                if let Err(e) = res {
+                    log::debug!("Failed to accept peer {peer_id}: {e}");
+                }
             }
-            .boxed(),
-            Command::SendMessage { peer, message } => async move {
-                boxed_cb(move |this| {
-                    let res = this.send_message(peer, message);
-                    if let Err(e) = res {
-                        log::debug!("Failed to send request to peer {peer}: {e}")
-                    }
-                    Ok(())
-                })
+            Command::Disconnect { peer_id } => {
+                let res = self.disconnect_peer(peer_id);
+                if let Err(e) = res {
+                    log::debug!("Failed to disconnect peer {peer_id}: {e}");
+                }
             }
-            .boxed(),
-            Command::AnnounceData { topic, message } => async move {
-                boxed_cb(move |this| {
-                    let res = this.announce_data(topic, message);
-                    if let Err(e) = res {
-                        log::error!("Failed to send announce data: {e}")
-                    }
-                    Ok(())
-                })
+            Command::SendMessage { peer, message } => {
+                let res = self.send_message(peer, message);
+                if let Err(e) = res {
+                    log::debug!("Failed to send request to peer {peer}: {e}")
+                }
+            }
+            Command::AnnounceData { topic, message } => {
+                let res = self.announce_data(topic, message);
+                if let Err(e) = res {
+                    log::error!("Failed to send announce data: {e}")
+                }
             }
-            .boxed(),
         };
+    }
 
-        self.command_queue.push(backend_task);
-
-        Ok(())
+    fn send_sync_event(sync_tx: &mpsc::UnboundedSender<SyncingEvent>, event: SyncingEvent) {
+        // SyncManager should always be active and so sending to a closed `conn_tx` is not a backend's problem, just log the error.
+        // NOTE: `sync_tx` is not connected in some PeerManager tests.
+        let res = sync_tx.send(event);
+        if res.is_err() {
+            log::error!("sending sync event from the backend failed unexpectedly");
+        }
```

Comment:
Why not propagate an error here?

Reply:
I think the code will be better if we ignore errors from closed channels (where it's possible).
If BlockSyncManager's channel is closed, then either the node is shutting down or BlockSyncManager has unexpectedly stopped.
If the node is shutting down, then the backend's even loop will terminate on the next iterations (once both BlockSyncManager and PeerManager are destroyed).
And if BlockSyncManager has stopped unexpectedly, then this should be handled appropriately where BlockSyncManager was started (I added TODO for this):
```
    tokio::spawn(async move {
        sync::BlockSyncManager::<T>::new(
            chain_config,
            p2p_config,
            sync,
            chainstate_handle,
            rx_p2p_sync,
            tx_peer_manager,
        )
        .run()
        .await
        .tap_err(|err| log::error!("SyncManager failed: {err}"))
    });
```